mono/metadata/marshal-ilgen.c

```diff
@@ -1193,6 +1201,16 @@ emit_thread_force_interrupt_checkpoint (MonoMethodBuilder *mb)
 	emit_thread_interrupt_checkpoint_call (mb, (gpointer)mono_thread_force_interruption_checkpoint_noraise);
 }
 
+MonoException *
+mono_marshal_rethrow_exception (MonoException *exc)
+{
+	mono_get_eh_callbacks ()->mono_reraise_exception (exc);
+
```

Comment:
I think we should add a new CEE_MONO_RETHROW opcode instead of this. mono_reraise_exception () is a trampoline which throws an exception and it can't be called from native code.